time_interp/include/time_interp_external2_bridge.inc

```diff
@@ -0,0 +1,295 @@
+!***********************************************************************
+!*                   GNU Lesser General Public License
+!*
+!* This file is part of the GFDL Flexible Modeling System (FMS).
+!*
+!* FMS is free software: you can redistribute it and/or modify it under
+!* the terms of the GNU Lesser General Public License as published by
+!* the Free Software Foundation, either version 3 of the License, or (at
+!* your option) any later version.
+!*
+!* FMS is distributed in the hope that it will be useful, but WITHOUT
+!* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+!* FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+!* for more details.
+!*
+!* You should have received a copy of the GNU Lesser General Public
+!* License along with FMS.  If not, see <http://www.gnu.org/licenses/>.
+!***********************************************************************
+!> @ingroup time_interp
+!> @addtogroup time_interp_external2_mod
+!> @{
+
+    !> @brief 2D interpolation for @ref time_interp_external_bridge
+    subroutine TIME_INTERP_EXTERNAL_BRIDGE_2D_(index1, index2, time, data_in, interp, verbose,horz_interp, mask_out, &
+               is_in, ie_in, js_in, je_in, window_id)
+
+      integer, intent(in) :: index1, index2
+      type(time_type), intent(in) :: time
+      real(FMS_TI_KIND_), dimension(:,:), intent(inout) :: data_in
+      integer, intent(in), optional :: interp
+      logical, intent(in), optional :: verbose
+      type(horiz_interp_type),intent(in), optional :: horz_interp
+      logical, dimension(:,:), intent(out), optional :: mask_out !< set to true where output data is valid
+      integer,                  intent(in), optional :: is_in, ie_in, js_in, je_in
+      integer,                  intent(in), optional :: window_id
+
+      real(FMS_TI_KIND_), dimension(size(data_in,1), size(data_in,2), 1) :: data_out
+      logical, dimension(size(data_in,1), size(data_in,2), 1) :: mask3d
```

Comment:
This is a new subroutine, so these variables should all be documented in doxygen-style